### PR TITLE
Sk types

### PR DESCRIPTION
**Pull request by embhorn**
Adding support for other stack object types in `sk_value()`

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -24561,13 +24561,52 @@ int  wolfSSL_sk_num(WOLF_STACK_OF(WOLFSSL_ASN1_OBJECT)* sk)
...
+
     for (; sk != NULL && i > 0; i--)
         sk = sk->next;
     if (sk == NULL)
         return NULL;
-    return (void*)sk->data.obj;
-}
 
+    switch (sk->type) {
+    #if defined(OPENSSL_ALL) || defined(WOLFSSL_QT)
```
Comment (JacobBarthelmeh) [inline src/ssl.c]: Some of these are available in OPENSSL_EXTRA without OPENSSL_ALL, only sk->data.access and sk->data.ext are limited to the OPENSSL_ALL/WOLFSSL_QT build.

File: src/ssl.c [hunk 2]
```diff
@@ -24561,13 +24561,52 @@ int  wolfSSL_sk_num(WOLF_STACK_OF(WOLFSSL_ASN1_OBJECT)* sk)
 
 void* wolfSSL_sk_value(WOLF_STACK_OF(WOLFSSL_ASN1_OBJECT)* sk, int i)
```
Comment (JacobBarthelmeh) [inline src/ssl.c]: With making the function generic to all stack types the signature of (WOLFSSL_STACK* sk, int i) may be better.

File: wolfssl/internal.h [hunk 3]
```diff
@@ -1842,9 +1842,18 @@ WOLFSSL_LOCAL int  SetCipherList(WOLFSSL_CTX*, Suites*, const char* list);
 #endif /* WOLFSSL_DTLS_EXPORT_TYPES */
 
 
+#if defined(OPENSSL_ALL) || defined(WOLFSSL_QT)
+#define UINT8_SZ 255
```
Comment (JacobBarthelmeh) [inline wolfssl/internal.h]: Naming of macro here could be adjusted. There is potential to clash with a UINT8_SZ macro set for the size of an unsigned byte. Maybe use MAX_SUITE_NAME from wolfssl/internal.h or MAX_CIPHERNAME_SZ  from wolfssl/internal.h or something equivalent.

File: src/ssl.c (hunk 2 above)
Comment (embhorn) [reply src/ssl.c]: Also updated wolfSSL_sk_num

File: src/ssl.c (hunk 1 above)
Comment (embhorn) [reply src/ssl.c]: Check

File: wolfssl/internal.h (hunk 3 above)
Comment (embhorn) [reply wolfssl/internal.h]: I went with `MAX_CIPHERNAME_SZ`

File: src/ssl.c (hunk 1 above)
Comment (embhorn) [reply src/ssl.c]: Also `sk->data.cipher->offset` and `WOLFSSL_GENERAL_NAME` are conditioned by OPENSSL_ALL/WOLFSSL_QT.

**Comment by embhorn** [comment]
Added support in wolfSSL_sk_value for STACK_OF(CONF_VALUE)

File: src/ssl.c [hunk 4]
```diff
@@ -25750,21 +25750,63 @@ void wolfSSL_ASN1_GENERALIZEDTIME_free(WOLFSSL_ASN1_TIME* asn1Time)
...
+            break;
+        case STACK_TYPE_NULL:
+            return (void*)sk->data.generic;
+            break;
+    #if defined(OPENSSL_ALL) || defined(WOLFSSL_QT)
+        case STACK_TYPE_X509_EXT:
+            return (void*)sk->data.ext;
+    #endif
+        case STACK_TYPE_CONF_VALUE:
+            return (void*)sk->data.conf->value;
```
Comment (dgarske) [inline src/ssl.c]: This provides a way to get the `conf->value`, but I see no place in the code where it can be set.

**Pull request merged**